src/main/java/com/github/manolo8/darkbot/core/objects/facades/SeassonPassMediator.java

```diff
@@ -0,0 +1,135 @@
+
+import org.jetbrains.annotations.Nullable;
+
+public class SeassonPassMediator extends Updatable implements SeassonPassAPI {
+
+    @Getter
+    private CurrentLevelProgress currentLevelProgress = new CurrentLevelProgress();
+
+    @Getter
+    private CurrentLevelProgress currentSeassonLevelInfo = new CurrentLevelProgress();
+
+    @Getter
+    private boolean seassonPassAvailable = false;
+
+    private final FlashList<SeassonPassQuest> dailyQuests = FlashList.ofVector(SeassonPassQuest.class);
+    private final FlashList<SeassonPassQuest> weeklyQuests = FlashList.ofVector(SeassonPassQuest.class);
+    private final FlashList<SeassonPassQuest> seassonQuests = FlashList.ofVector(SeassonPassQuest.class);
+    private boolean allQuestsUpdated = false;
+    private final FlashList<SeassonPassQuest> allQuests = FlashList.ofVector(SeassonPassQuest.class);
+
+    private long questDataAddr = 0;
+
+    @Override
+    public void update() {
+        if (address == 0) {
+            return;
+        }
+
+        this.allQuestsUpdated = false;
+
+        long data = API.readAtom(address + 0x60);
+        this.seassonPassAvailable = API.readBoolean(API.readAtom(data + 0x70) + 0x20);
+
+        if (!seassonPassAvailable) {
+            return;
+        }
+
+        this.currentLevelProgress.update(API.readAtom(data + 0x58));
+        this.currentSeassonLevelInfo.update(API.readAtom(data + 0x68));
+        this.questDataAddr = API.readAtom(address + 0x78);
+
+        if (questDataAddr == 0) {
+            return;
+        }
+
+        dailyQuests.update(API.readAtom(questDataAddr + 0x60));
+        weeklyQuests.update(API.readAtom(questDataAddr + 0x68));
+        seassonQuests.update(API.readAtom(questDataAddr + 0x70));
+    }
+
+    public @Nullable List<? extends SeassonPassQuest> getDailyQuests() {
+        return this.dailyQuests;
+    }
+
+    public @Nullable List<? extends SeassonPassQuest> getWeeklyQuests() {
+        return this.weeklyQuests;
+    }
+
+    public @Nullable List<? extends SeassonPassQuest> getSeassonQuests() {
+        return this.seassonQuests;
+    }
+
+    public @Nullable List<? extends SeassonPassQuest> getAllQuests() {
+        if (!allQuestsUpdated && questDataAddr != 0) {
+            allQuestsUpdated = true;
+            allQuests.update(API.readAtom(questDataAddr + 0x58));
+        }
+
+        return this.allQuests;
+    }
+
+    @Getter
+    @ToString
+    public static class CurrentLevelProgress extends Auto implements SeassonPassAPI.CurrentLevelProgress {
+        private int maxProgress;
+        private int currentProgress;
+
+        @Override
+        public void update() {
+            if (address == 0) {
+                return;
+            }
+
+            this.maxProgress = API.readInt(address + 0x24);
+            this.currentProgress = API.readInt(address + 0x28);
+        }
+
+        public double getProgressPercentage() {
+            return this.currentProgress / this.maxProgress;

```

Comment:
If max progress wasn't ever read, you're going to divide by 0
Also you're doing integer division and returning it as a double, this will always be 0.

Consider using an IDE which has warnings for things like these like intellij
![image](https://github.com/darkbot-reloaded/DarkBot/assets/11789291/48dbde15-acf3-4ba6-9e6b-6f99341e6825)


```suggestion
            return maxProgress == 0 ? 1d : (double) this.currentProgress / this.maxProgress;
```

Reply:
done